sdks/python/apache_beam/runners/pipeline_context.py

```diff
@@ -49,6 +50,32 @@
   from apache_beam.coders.coder_impl import IterableStateWriter
 
 
+class ComponentIdContext(object):
```

Comment:
There's a lot of overlap between the purpose of this and the purpose of the `PipelineContext` but I can see how they are different. I see that the pipeline only takes the context at `to_runner_api` time, and the context has tweaks like `use_fake_coders` and `default_environment`. So when using xlang expansion you can keep just the ids and throw away the contents that will be generated according to these tweaks later.